arlo-client/src/components/Audit/Setup/Contests/index.tsx

```diff
@@ -74,7 +74,7 @@ const Contests: React.FC<IProps> = ({
     // TEST TODO
     /* istanbul ignore next */
     if (!response) return
-    nextStage.activate()
+    nextStage.activate!()
```

Comment:
what's the reasoning for `activate` to be possibly null?

Reply:
@benadida Because in the sidebar for the JA setup flow, there's literally only one screen, so only one item in the menu, therefore no toggling between them. Thus no `onClick` function that would ever do anything. Instead of doing a null function that we'd then have to test that it did nothing, just excluding it made more sense to me. We always know that it exists in the AA setup sidebar, though, so we can just tell TS that.

Reply:
I think the problem here is that, if `activate` is null, this will compile okay, but will fail at runtime. The `!`, if I understand it correctly (and I'm relatively new to it as of last year), is just telling the compiler to shut up and ignore the potential null case, but it doesn't change the runtime.

So if `activate` can be null, we probably want this instead:

```
if (nextStage.activate) nextStage.activate()
```

Reply:
I see your point. My concern is that the only time it would get a null value at runtime would be if someone accidentally coded it to pass in the wrong variable. In which case it *should* fail at runtime and get fixed, instead of being handled gracefully. If it simply doesn't call the function if the function doesn't exist, that would actually be a bug that could possibly be missed. Just my thought on it, though.

Reply:
got it. My sense is that there might then be a better way to model it, but in the interest of moving forward, if this is meant to fail on null, then I can live with this.

Small suggestion: raise an explicit error that logs what happened?

Reply:
I could do that, though in the interests of saving on writing tests, I'll add an istanbul ignore comment too, haha.

Reply:
for sure. And to be clear I mean very little investment here, a UI if you already have it (toast?), or just a `console.log`.